pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -971,14 +972,18 @@ def getDisplayDataset(self):
 
         if self.opts['autoDownsample']:
             # this option presumes that x-values have uniform spacing
-            if view_range is not None and len(x) > 1:
-                dx = float(x[-1]-x[0]) / (len(x)-1)
+
+            finite_x = x[np.isfinite(x)]  # ignore infinite and nan values
+            if view_range is not None and len(finite_x) > 1:
+                dx = float(finite_x[-1]-finite_x[0]) / (len(finite_x)-1)
                 if dx != 0.0:
-                    x0 = (view_range.left()-x[0]) / dx
-                    x1 = (view_range.right()-x[0]) / dx
+                    x0 = (view_range.left()-finite_x[0]) / dx
+                    x1 = (view_range.right()-finite_x[0]) / dx
                     width = self.getViewBox().width()
-                    if width != 0.0:
-                        ds = int(max(1, int((x1-x0) / (width*self.opts['autoDownsampleFactor']))))
+                    if width != 0.0:  # autoDownsampleFactor _should_ be > 1.0
```

Comment:
Hi @StSav012 

I think for this equality comparison to 0.0, I think we should use `math.isclose(width, 0.0, abs_tol=1e-9)`

Reply:
With a `not` in front of it.

Reply:
Hi, @j9ac9k @outofculture 

I can't agree with the approach you propose. Not yet, at least.

The check for infinity takes care of the cases when `width` is not exactly zero, but close to it. I'm afraid, with `isclose`, the behavior would not be straightforward; it would depend on the tolerance picked essentially at random. Say, if the `x` span is huge (like `1e308`), we might still get the infinity, even if `width` is significantly non-zero. However, if `width` is exactly `0.0`, the function fails with `ZeroDivisionError`, and the rest of the function gets ignored. On the other hand, if `view_range.right() - view_range.left()` (why do we need `x0` and `x1` at all?) is small (say, equal to the `isclose` tolerance), the picked tolerance might be far too rough.

Reply:
My understanding is that `width` here wasn't in the view range coordinates, but corresponds to `QGraphicsWidget.geometry().width()` which should return the `QRectF` that represents the position and size of the widet, and not the view space. 

That said, the tolerance picked effectively at random so you're not wrong there...

Reply:
Hmm. Looking further, `QRect.width()` should be an [integer](https://doc.qt.io/qtforpython/PySide6/QtCore/QRect.html#PySide6.QtCore.PySide6.QtCore.QRect.width), so I don't think this protection will ever matter either way. We could wrap it in a `try`/`except ZeroDivisionError: pass`, even, if that's all we're guarding against, with the added possible bonus of protecting against `self.opts['autoDownsampleFactor'] == 0`, too. That leaves the question of whether we need to treat e.g. `1 / ((0.3 - 0.2) - 0.1)` as a divide-by-zero, which we would normally want to do. In this case, if a application sets `self.opts['autoDownsampleFactor']` to e.g. 1e-15, they're going to get bad behavior ("please _up-sample_ my data"?), so we can probably safely ignore this aspect of the problem. Our knee-jerk resistance to comparing floats to specific values likely doesn't matter either way here, but the sentiment is a good one, so it might be worthwhile to put in the `isclose` just to uphold that principle.

Reply:
> Hmm. Looking further, `QRect.width()` should be an [integer](https://doc.qt.io/qtforpython/PySide6/QtCore/QRect.html#PySide6.QtCore.PySide6.QtCore.QRect.width), so I don't think this protection will ever matter either way. 


`QRect.width()` should absolutely be an integer, but we're dealing w/ a `QRectF` here ;)

https://doc-snapshots.qt.io/qt6-dev/qgraphicslayoutitem.html#geometry

> We could wrap it in a `try`/`except ZeroDivisionError: pass`, even, if that's all we're guarding against, with the added possible bonus of protecting against `self.opts['autoDownsampleFactor'] == 0`, too. 

nit-pick level 100; if we do that we should use 

```python
import contextlib

...

with contextlib.suppress(ZeroDivisionError):
    # do the division here
```

> That leaves the question of whether we need to treat e.g. `1 / ((0.3 - 0.2) - 0.1)` as a divide-by-zero, which we would normally want to do. In this case, if a application sets `self.opts['autoDownsampleFactor']` to e.g. 1e-15, they're going to get bad behavior ("please _up-sample_ my data"?), so we can probably safely ignore this aspect of the problem. Our knee-jerk resistance to comparing floats to specific values likely doesn't matter either way here, but the sentiment is a good one, so it might be worthwhile to put in the `isclose` just to uphold that principle.

Yeah, @StSav012 is modifying existing code here that has some code I absolutely hate, and I'm requesting they change something else that was already in the library; probably not the most fair ask.  I'm good w/ leaving it for this PR, and addressing the float equality comparisons in another PR.